src/topology/continuous_function/algebra.lean

```diff
@@ -811,35 +812,39 @@ We now provide formulas for `f ⊓ g` and `f ⊔ g`, where `f g : C(α, β)`,
 in terms of `continuous_map.abs`.
 -/
 
-section
-variables {R : Type*} [linear_ordered_field R]
+namespace continuous_map
 
--- TODO:
--- This lemma (and the next) could go all the way back in `algebra.order.field`,
--- except that it is tedious to prove without tactics.
--- Rather than stranding it at some intermediate location,
--- it's here, immediately prior to the point of use.
-lemma min_eq_half_add_sub_abs_sub {x y : R} : min x y = 2⁻¹ * (x + y - |x - y|) :=
-by cases le_total x y with h h; field_simp [h, abs_of_nonneg, abs_of_nonpos, mul_two]; abel
+section lattice
+variables {α : Type*} [topological_space α]
+variables {β : Type*} [topological_space β]
 
-lemma max_eq_half_add_add_abs_sub {x y : R} : max x y = 2⁻¹ * (x + y + |x - y|) :=
-by cases le_total x y with h h; field_simp [h, abs_of_nonneg, abs_of_nonpos, mul_two]; abel
+/- C(α, β) is a lattice ordered group -/
 
+@[to_additive continuous_map.has_le.le.add_covariant_class]
+instance [lattice β] [comm_group β] [topological_group β] [covariant_class β β (*) (≤)] :
+  covariant_class C(α, β) C(α, β) (*) (≤) :=
+begin
+    refine ⟨λ f g₁ g₂ hg₁₂, _⟩,
+    intro x,
+    simp only [to_fun_eq_coe, mul_apply, mul_le_mul_iff_left],
+    apply hg₁₂,
 end
 
-namespace continuous_map
+variables [linear_ordered_field β] [topological_ring β] [order_topology β]
 
-section lattice
-variables {α : Type*} [topological_space α]
-variables {β : Type*} [linear_ordered_field β] [topological_space β]
-  [order_topology β] [topological_ring β]
+open lattice_ordered_comm_group
 
 lemma inf_eq (f g : C(α, β)) : f ⊓ g = (2⁻¹ : β) • (f + g - |f - g|) :=
-ext (λ x, by simpa using min_eq_half_add_sub_abs_sub)
+begin
+  rw [← inv_smul_eq_iff₀, inv_inv, abs_neg_comm, ← two_inf_eq_add_sub_abs_sub, two_smul, two_smul],
+  finish,
```

Comment:
Usually we try to avoid `finish`; what's the goal here, and can `simp` close it?

Reply:
@eric-wieser I find it frustratingly hard to convince Mathlib that `(2 : K) •a` and `(2 : ℕ) •a` are the same thing.

e.g.
```
variables (M : Type*)  [add_comm_group M] [lattice M] [covariant_class M M (+) (≤)]
variables (K : Type*) [division_ring K] [module K M]

lemma min_eq_half_add_sub_abs_sub  {a b : M} :
  a ⊓ b = (2⁻¹ : K) • (a + b - |b - a|) :=
begin
  rw [← two_inf_eq_add_sub_abs_sub, ← inv_smul_eq_iff₀, inv_inv],
```
Gets me to the goal
```
2 • (a ⊓ b) = 2 • (a ⊓ b)
```
Which I think is actually:
```
(2 : K) • (a ⊓ b) = (2 : ℕ) • (a ⊓ b)
```
I usually spend hours going round in circles when I get to this point.

Reply:
@eric-wieser Perhaps we need to exclude division rings of characteristic 2?

Reply:
Possibly we want `variables (K : Type*) [ring K] [invertible (2 : K)] [module K M]` - and maybe some sort of scalar tower?

Reply:
Sorry, this is beyond my grasp of algebra.

Reply:
Thanks for trying, I think that's pretty much on the right track. I'll have a go shortly.

Reply:
I've pushed the lemmas I was thinking of

Reply:
I assume `inf_mem_subalgebra_closure` should now look like this?

https://github.com/leanprover-community/mathlib/pull/18780/commits/760eb326194f11f587946c3ae6d1e89164919ef2#diff-9762e2b1c94ba40d1d004173d980b2889bee3d33b2bc4358ea09f98fd4b8af74L124 ?

I am not sure why `covariant_class C(X, ℝ) C(X, ℝ) has_add.add has_le.le` isn't automatic? I tried to investigate, but everything I try leads to `excessive memory consumption`.